lib/ivis_opengl/gfx_api_gl.cpp

```diff
@@ -62,6 +76,38 @@ struct gl_texture : public gfx_api::texture
 
 };
 
+struct gl_buffer : public gfx_api::buffer
+{
+	gfx_api::buffer::usage usage;
+	GLuint buffer;
+
+	virtual ~gl_buffer() override
+	{
+		glDeleteBuffers(1, &buffer);
+	}
+
+	gl_buffer(const gfx_api::buffer::usage& usage, const size_t& size)
+		: usage(usage)
+	{
+		glGenBuffers(1, &buffer);
+		glBindBuffer(to_gl(usage), buffer);
+		glBufferData(to_gl(usage), size, nullptr, GL_STATIC_DRAW);
```

Comment:
You always pass `GL_STATIC_DRAW` here, but the original code uses different values (`GL_STATIC_DRAW`, `GL_STREAM_DRAW`, and `GL_DYNAMIC_DRAW`).